src/ClientBuilder.php

```diff
@@ -321,13 +321,13 @@ private function createTransportInstance(): TransportInterface
                 throw new \RuntimeException('The `http_proxy` option does not work together with a custom client.');
             }
 
-            if (HttpAsyncClientDiscovery::safeClassExists(HttpCurlClient::class)) {
-                $this->httpClient = new HttpCurlClient(null, null, [
-                    CURLOPT_PROXY => $this->options->getHttpProxy(),
-                ]);
-            } else {
+            if (class_exists(HttpCurlClient::class)) {
```

Comment:
Shouldn't this be `!` 🤔 ?
```suggestion
            if (!class_exists(HttpCurlClient::class)) {
```

Reply:
Good catch!